sources/api/bootstrap-containers/README.md

```diff
@@ -0,0 +1,61 @@
+# bootstrap-containers
+
+Current version: 0.1.0
+
+## Bootstrap containers
+
+Bootstrap containers are host containers that can be used to setup the host
+before services like the ecs-agent, kubelet and docker start. The [configured](../#configuredtarget)
+target depends on the systemd units created for this type of containers, thus
+blocking the init process until all the bootstrap containers exit.
+
+This type of containers are very similar to normal host containers, they come with
+persistent storage and with optional user data. Unlike normal host containers, bootstrap
+containers can't be threated as `superpowered`. However, these containers run as privileged
+and have access to the underlying root filesystem on `/.bottlerocket/rootfs`.
+
+Bootstrap containers have three different modes:
+
+* `always`: with this setting, the container is executed in all boots.
+* `off`: blocks the execution of the container
+* `once`: with this setting, the container only runs on the first boot. Upon completion,
+the container marks itself as `off`. If you need to execute the containers after the first boot,
+you can mark it as `once` and reboot.
+
+## Examples
+Given a bootstrap container with the following configuration:
+
+```toml
+[settings.bootstrap-containers.bear]
+source="<SOURCE>"
+mode="once"
+user-data="ypXCt82h4bSlwrfKlA=="
+```
+
+Where `<SOURCE>`, is the url of an image with the following definition:
+
+```Dockerfile
+FROM alpine
+ADD bootstrap-script /
+RUN chmod +x /bootstrap-script
+ENTRYPOINT ["sh", "bootstrap-script"]
+```
+
+And `bootstrap-script` as:
+
+```sh
+#!/bin/sh
+BEAR_DIR=/.bottlerocket/bootstrap-containers/bear
+cat $BEAR_DIR/user-data
+sleep 120
```

Comment:
In this example, will boot be delayed by 120s until this completes?

Reply:
Yes, that's correct. With this bootstrap container, systemd will "wait" for the container to exit. Upon completion, the boot process will continue.

Reply:
What happens if the container exits with a failure code?

Reply:
Hi @diranged sorry for the late response. We added the flag `essential`, it can be set to `true` to fail the boot if the underlying bootstrap container exited with non-zero.